src/API/Site/Controllers/MerchantCenter/ShippingRateController.php

```diff
@@ -175,25 +184,51 @@ protected function get_suggestions_callback(): callable {
 	 */
 	protected function get_read_rate_callback(): callable {
 		return function ( Request $request ) {
-			$country = $request->get_param( 'country_code' );
-			$rates   = $this->get_shipping_rates_for_country( $country );
-			if ( empty( $rates ) ) {
+			$id   = (string) $request->get_param( 'id' );
+			$rate = $this->get_shipping_rate_by_id( $id );
+			if ( empty( $rate ) ) {
 				return new Response(
 					[
 						'message' => __( 'No rate available.', 'google-listings-and-ads' ),
-						'country' => $country,
+						'id'      => $id,
+					],
+					404
+				);
+			}
+
+			return $this->prepare_item_for_response( $rate, $request );
+		};
+	}
+
+	/**
+	 * @return callable
+	 *
+	 * @since x.x.x
+	 */
+	protected function get_update_rate_callback(): callable {
+		return function ( Request $request ) {
+			$id = (string) $request->get_param( 'id' );
```

Comment:
Is there a particular reason we need to cast it to a string (same is done for the delete callback)? In the schema it's a number and in the DB it's an int, wouldn't it be better to modify the function `get_shipping_rate_by_id` to accept an int?

Reply:
Interesting, I don't recall taking this into consideration for other tables. I always had the impression since WP uses a bigint for storing post ID's and treats it in PHP as an int we should be alright to do the same for our other tables. WP tables use an even larger range since it's an unsigned bigint. I guess in theory it could be a problem for some platforms that run on 32 bit (but probably not very common).